src/plugins/discover/public/application/view_components/canvas/top_nav.tsx

```diff
@@ -69,6 +69,16 @@ export const TopNav = ({ opts, showSaveQuery, isEnhancementsEnabled }: TopNavPro
     uiSettings
   );
 
+  useEffect(() => {
+    const subscription = data$.subscribe((queryData) => {
+      data.ui.setQueryStatus({
+        status: queryData.status,
+        message: queryData.statusBody?.errorMsg,
+        time: queryData.statusBody?.time,
+      });
+    });
```

Comment:
do we need to cleanup subscription on unmount?

Reply:
returned a unsubscribe()